dali/pipeline/operator/common.h

```diff
@@ -38,5 +43,130 @@ inline void GetSingleOrRepeatedArg(const OpSpec &spec, vector<T> &result,
       to_string(result.size()) + " given.");
 }
 
+template <typename T>
+void GetPerSampleArgument(std::vector<T> &output,
+                          const std::string &argument_name,
+                          const OpSpec &spec,
+                          const ArgumentWorkspace &ws,
+                          int batch_size = -1 /* -1 = get from "batch_size" arg */) {
+  if (batch_size < 0)
+    batch_size = spec.GetArgument<int>("batch_size");
+
+  if (spec.HasTensorArgument(argument_name)) {
+    const auto &arg = ws.ArgumentInput(argument_name);
+    decltype(auto) shape = arg.shape();
+    int N = shape.num_samples();
+    if (N == 1) {
+      bool is_valid_shape = shape.tensor_shape(0) == TensorShape<1>{batch_size};
+
+      DALI_ENFORCE(is_valid_shape,
+        make_string("`", argument_name, "` must be a 1xN or Nx1 (N = ", batch_size,
+                    ") tensor list. Got: ", shape));
+
+      output.resize(batch_size);
+      auto *data = arg[0].template data<T>();
+
+      for (int i = 0; i < batch_size; i++) {
+        output[i] = data[i];
+      }
+    } else {
+      bool is_valid_shape = N == batch_size &&
+                            is_uniform(shape) &&
+                            volume(shape.tensor_shape_span(0)) == 1;
+      DALI_ENFORCE(is_valid_shape,
+        make_string("`", argument_name, "` must be a 1xN or Nx1 (N = ", batch_size,
+                    ") tensor list. Got: ", shape));
+
+      output.resize(batch_size);
+      for (int i = 0; i < batch_size; i++) {
+        output[i] = arg[i].template data<T>()[0];
+      }
+    }
+  } else {
+    output.clear();
+    output.resize(batch_size, spec.GetArgument<T>(argument_name));
+  }
+  assert(output.size() == static_cast<size_t>(batch_size));
+}
+
+
+template <typename ArgumentType, typename ExtentType>
+int GetShapeLikeArgument(std::vector<ExtentType> &out_shape,
+                         const OpSpec &spec,
+                         const std::string &argument_name,
+                         const ArgumentWorkspace &ws,
+                         int ndim = -1,
+                         int batch_size = -1 /* -1 = get from "batch_size" arg */) {
+  if (batch_size < 0)
+    batch_size = spec.GetArgument<int>("batch_size");
+
+  auto to_extent = [](ArgumentType extent) {
+    return std::is_floating_point<ArgumentType>::value && std::is_integral<ExtentType>::value
+      ? static_cast<ExtentType>(std::lround(extent))
+      : static_cast<ExtentType>(extent);
+  };
+
+  if (spec.HasTensorArgument(argument_name)) {
+    const auto &arg = ws.ArgumentInput(argument_name);
+    auto argview = view<const ArgumentType>(arg);
+    int N = argview.shape.num_samples();
+    DALI_ENFORCE(N == batch_size, make_string("Unexpected number of samples in argument `",
```

Comment:
```suggestion
    DALI_ENFORCE(N >= batch_size, make_string("Unexpected number of samples in argument `",
```
In the future when we will provide the ability to change batch size I think the user need to provide at least as much args as sample sin the batch, and it should be okay to provide more.

Reply:
Since batch_size is provided externally, it needs to be exactly that.